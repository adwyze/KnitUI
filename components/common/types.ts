export interface IStyled<PropType> {
  customProps: PropType
  theme: any
  [htmlProp: string]: any
}
