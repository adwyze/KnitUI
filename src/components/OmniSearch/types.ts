import {
  BaseComponentProps,
} from "../../common/types"
import React, { ReactNode } from "react"

export interface OmniSearchProps extends BaseComponentProps {
  /** Whether the search box is visible or not  */
  visible: boolean

  /** The value to be seen in the input box */
  value: string

  /** placeholder string when the value is an empty string */
  placeholder?: string

  /** The dropdown component that is to be rendered below the search input box */
  dropdown?: ReactNode | null

  /** The callback made when the user changes the input */
  onChange: (e: React.FormEvent<HTMLInputElement>) => void

  /** handler that is run when close is initiated (eg. click outside) */
  onClose: () => void
}
