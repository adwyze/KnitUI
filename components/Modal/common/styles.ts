import * as theme from "../../styles/variables"

export const border = `1px solid ${theme.shades.gray85}`
export const borderRadius = "0.4rem"
export const padding = {
  vertical: "2.1rem",
  horizontal: "2.8rem",
}
