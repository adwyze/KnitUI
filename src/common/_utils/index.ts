import { ThemedStyledFunction } from "styled-components"
import chroma from "chroma-js"
import { ColorPreset, CustomColor } from "../types"

export const insertIf = (
  obj: any = {},
  condition: boolean,
  isArray: boolean = false
) => {
  if (condition) {
    return obj
  }
  if (isArray) {
    return []
  }
  return {}
}

// https://github.com/styled-components/styled-components/issues/630
export function withProps<U>() {
  return <
    P extends keyof JSX.IntrinsicElements | React.ComponentType<any>,
    T extends object,
    O extends object = {}
  >(
    fn: ThemedStyledFunction<P, T, O>
  ): ThemedStyledFunction<P & U, T, O & U> =>
    (fn as unknown) as ThemedStyledFunction<P & U, T, O & U>
}

// Color utils

/**
 * Checks if the color is light or darks based on its L value in
 * the HSL color scale.
 * @param color A chroma color instance
 */
export const isLightColor = color => color.get("hsl.l") >= 0.5

/**
 * Returns font color of updated lightness on hover
 * @param color A chroma color instance
 */
export const getHoverFontColor = color => {
  return isLightColor(color)
    ? color.set("hsl.l", "-0.3")
    : color.set("hsl.l", "+0.3")
}

/**
 * Returns background color of updated lightness on hover
 * @param color A chroma color instance
 */
export const getHoverBackgroundColor = color => {
  return isLightColor(color)
    ? color.set("hsl.l", "-0.1")
    : color.set("hsl.l", "+0.1")
}


/**
 * Determines the appropriate font color based on the background color.
 */
export const getFontColor = (theme, backGroundColor) => {
  return isLightColor(backGroundColor)
    ? theme.chromaPalette.Neutral90
    : theme.chromaPalette.Neutral0
}

const createParsedColorTheme = (theme, backgroundColor) => ({
  background: backgroundColor,
  font: getFontColor(theme, backgroundColor),
})

/**
 * Creates an object with background and font properties initialized
 * with appropriate chroma colors.
 * @param backGroundColor A plain CSS color string
 */
export const parseCustomColor = (theme, customColor: CustomColor) => {
  if (
    !(
      chroma.valid(customColor) ||
      (typeof customColor === "object" &&
        chroma.valid(customColor.color) &&
        chroma.valid(customColor.secondaryColor))
    )
  ) {
    throw new Error("Provided color theme contains invalid color values")
  }
  if (typeof customColor === "object") {
    return {
      font: chroma(customColor.secondaryColor),
      background: chroma(customColor.color),
    }
  }
  customColor = chroma(customColor)
  return createParsedColorTheme(theme, customColor)
}

/**
 * Creates an object with background and font properties initialized
 * with appropriate chroma colors.
 * @param backGroundColor One of the predefined preset values
 */
export const parseColorPreset = (theme, backgroundColor: ColorPreset) => {
  backgroundColor = theme.secondaryPalette[backgroundColor]
  return createParsedColorTheme(theme, backgroundColor)
}

/**
 * Get Color from Styled Component Props
 */
export const getThemeColor = (props: any, color: string) => {
  const {
    theme: {
      knitui: { chromaPalette },
    },
  } = props

  return chromaPalette[color].hex()
}

/**
 * Get OS of current platform
 */
export const getOSName = () => {
  const platform = window.navigator.platform
  const userAgent = window.navigator.userAgent

  const macOSList = ["Macintosh", "MacIntel", "MacPPC", "Mac68K"]
  const windowOSList = ["Windows", "Win16", "Win32", "WinCE"]
  const iOSList = ["iPhone", "iPad", "iPod"]

  let os: string | undefined

  if (macOSList.includes(platform)) {
    os = "MacOS"
  } else if (windowOSList.includes(platform)) {
    os = "Windows"
  } else if (iOSList.includes(platform)) {
    os = "iOS"
  } else if (/Android/.test(userAgent)) {
    os = "Android"
  } else if (/Linux/.test(userAgent)) {
    os = "Linux"
  }

  return os
}
