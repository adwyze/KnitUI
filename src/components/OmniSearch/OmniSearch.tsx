import React, { Component, RefObject } from "react"
import styled from "styled-components"
import Dialog from "rc-dialog"
import Icon from "../Icon"
import { OmniSearchProps } from "./types"

import "rc-dialog/assets/index.css"

const StyledDialog = styled(Dialog)`
  .rc-dialog-body {
    padding: 7px 0;
  }
`

const InputWrapper = styled.div`
  padding: 10px;
  display: flex;
  margin: 7px 14px;
  align-items: center;
  &:focus-within {
    border: 1px solid ${props => getColor(props, "Azure80")};
    box-shadow: 0px 0px 2px #0066ff;
    border-radius: 4px;
  }
`

const StyledInput = styled.input`
  width: 100%;
  border: none;
  font-size: 1.6rem;
  margin-left: 1rem;
  height: 2rem;
  &::placeholder {
    color: ${props => getColor(props, "Neutral45")};
  }
`

const getColor = (props:any, color: string) => {
  const { theme: { knitui } } = props
  return knitui.chromaPalette[color]
}

class OmniSearch extends Component<OmniSearchProps> {
  searchRef: RefObject<HTMLInputElement>
  constructor(props: OmniSearchProps) {
    super(props)
    this.searchRef = React.createRef<HTMLInputElement>()
  }

  focusSearch = () => this.searchRef.current && this.searchRef.current.focus()

  componentDidUpdate(prevProps: OmniSearchProps) {
    const { visible } = this.props
    if (visible && !prevProps.visible) {
      this.focusSearch()
    }
  }

  render() {
    const {
      visible,
      onClose,
      onChange,
      value = "",
      dropdown = null,
      placeholder = "Search for a report",
    } = this.props

    return (
      <StyledDialog visible={visible} onClose={onClose} closable={false}>
        <InputWrapper>
          <Icon type={"oSearch"} size={"1.8rem"} fill="#A6A6A6" />
          <StyledInput
            type="search"
            role="search"
            autoFocus
            value={value}
            onChange={onChange}
            onBlur={this.focusSearch}
            placeholder={placeholder}
            ref={this.searchRef}
          />
        </InputWrapper>
        {dropdown}
      </StyledDialog>
    )
  }
}

export default OmniSearch
