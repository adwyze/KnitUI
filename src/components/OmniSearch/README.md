<!-- STORY -->

# OmniSearch

### Usage

```javascript
import { OmniSearch } from "KnitUI"

const OmniSearchWrapper = () => {
  const [visible, setVisible] = useState(true)
  const data = ["List Item 1", "List Item 2", "List Item 3"]
  const [value, setValue] = useState("")
  const [filteredData, setFilteredData] = useState(data)

  const ListItems = filteredData.map(item => <ListItem>{item}</ListItem>)
  return (
    <div>
      <Container
        style={{
          marginBottom: 20,
          padding: 20,
        }}>
        <div
          onClick={() => {
            setVisible(true)
          }}>
          <Button label={"Spotlight"} />
        </div>
        <OmniSearch
          visible={visible}
          value={value}
          onChange={event => {
            const value = event.target.value
            const filteredData = data.filter(item => item.includes(value))
            setValue(value)
            setFilteredData(filteredData)
          }}
          onClose={() => {
            setVisible(!visible)
          }}
          dropdown={ListItems}></OmniSearch>
      </Container>
    </div>
  )
}
```

### Props

| Prop name | Type      | Optional | Default | Description                                        |
| --------- | --------- | -------- | ------- | -------------------------------------------------- |
| visible   | boolean   | No       | None    | Toggles the visibility of the spotlight OmniSearch |
| value     | string    | No       | None    | This is a value of the input                       |
| onChange  | function  | No       | None    | The onChange function supplied to controlled input |
| onClose   | function  | No       | None    | Fires when the spotlight dialog is being closed    |
| dropdown  | ReactNode | Yes      | None    | content to be shown in the list                    |

**Note: Default HTML attributes of the input element are supported.**
