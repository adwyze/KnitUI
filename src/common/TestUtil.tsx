import React, { ReactElement } from "react"
import { render } from "react-testing-library"
import { ThemeProvider } from "./styles"

const ProviderWrapper = ({
  children,
}: {
  children: ReactElement
}): ReactElement => {
  // Nest all providers here
  return <ThemeProvider>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: object) =>
  render(ui, { wrapper: ProviderWrapper, ...options })

// re-export everything
export * from "react-testing-library"
// override render method
export { customRender as render }
