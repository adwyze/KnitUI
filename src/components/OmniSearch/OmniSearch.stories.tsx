import React, { useState } from "react"
import { storiesOf } from "@storybook/react"
import styled from "styled-components"
import { OmniSearch, Button } from "../"
import { getThemeColor } from "../../common/_utils"
import { withKnobs } from "@storybook/addon-knobs"
import { oAccountCircle as UserProfileIcon } from "../Icon/Icons"
const Readme = require("./README.md")

const stories = storiesOf("Omni Search", module)
stories.addDecorator(withKnobs)

const ListItem = styled.div`
  width: 100%;
  padding: 0.7rem 1.4rem;
  font-size: 1.4rem;
  line-height: 2rem;

  &:hover {
    background: ${props => getThemeColor(props, "Neutral10")};
  }
`

const Container = styled.div`
  border: 1px solid lightgray;
  height: 360px;
  width: 100%;
  font-size: 1.4rem;
`

const OmniSearchWrapper = () => {
  const [visible, setVisible] = useState(true)
  const data = [
    "List Item 1",
    "List Item 2",
    "List Item 3",
    "List Item 4",
    "List Item 5",
  ]
  const [value, setValue] = useState("")
  const [filteredData, setFilteredData] = useState(data)

  const ListItems = filteredData.map(item => <ListItem>{item}</ListItem>)
  return (
    <div>
      <Container
        style={{
          marginBottom: 20,
          padding: 20,
        }}>
        <div
          onClick={() => {
            setVisible(true)
          }}>
          <Button label={"Spotlight"} />
        </div>
        <OmniSearch
          visible={visible}
          value={value}
          onChange={event => {
            const value = event.target.value
            const filteredData = data.filter(item => item.includes(value))
            setValue(value)
            setFilteredData(filteredData)
          }}
          onClose={() => {
            setVisible(!visible)
          }}
          dropdown={ListItems}></OmniSearch>
      </Container>
    </div>
  )
}

stories
  .addParameters({
    readme: {
      // Show readme before story
      codeTheme: "shades-of-purple",
      // Show readme at the addons panel
      sidebar: Readme,
    },
  })
  .add("basic", () => <OmniSearchWrapper />)
