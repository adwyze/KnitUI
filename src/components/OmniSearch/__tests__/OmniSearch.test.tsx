import React from "react"
import OmniSearch from "../"
import { render } from "../../../common/TestUtil"
import { cleanup, fireEvent } from "@testing-library/react"

const baseProps = {
  visible: true,
  value: "demo",
  onChange: () => null,
  onClose: () => null,
}

describe("OmniSearch", () => {
  afterEach(cleanup)

  it("renders correctly when visible is false", () => {
    const { baseElement } = render(
      <OmniSearch {...baseProps} visible={false} />
    )
    expect(baseElement).toMatchSnapshot()
  })

  it("renders correctly when visible is true", () => {
    const { baseElement } = render(<OmniSearch {...baseProps} />)
    expect(baseElement).toMatchSnapshot()
  })

  it("is focused on render", () => {
    const { getByRole } = render(<OmniSearch {...baseProps} />)
    const input = getByRole("search")
    expect(document.activeElement).toBe(input)
  })

  it("maintains focus on the search input box", () => {
    const { container, getByRole } = render(<OmniSearch {...baseProps} />)
    // click on the padding outside the input...
    fireEvent.click(container)
    const input = getByRole("search")
    expect(document.activeElement).toBe(input)
  })

  it("renders correctly when dropdown is provided", () => {
    const component = <div data-testid="omnisearch-dropdown" />
    const { baseElement, queryByTestId } = render(
      <OmniSearch {...baseProps} dropdown={component} />
    )
    expect(baseElement).toMatchSnapshot()
    expect(queryByTestId("omnisearch-dropdown")).toBeTruthy()
  })

  it("triggers props.onChange correctly", () => {
    const onChangeMock = jest.fn()
    const { getByRole } = render(
      <OmniSearch {...baseProps} onChange={onChangeMock} />
    )
    const input = getByRole("search")
    fireEvent.change(input, { target: { value: "hello" } })
    expect(onChangeMock).toHaveBeenCalled()
  })

  it("uses the value and placeholder props correctly", () => {
    const { getByRole } = render(
      <OmniSearch {...baseProps} value="hello" placeholder="placeholder" />
    )
    const input = getByRole("search") as HTMLInputElement
    expect(input.value).toEqual("hello")
    expect(input.placeholder).toEqual("placeholder")
  })
})
