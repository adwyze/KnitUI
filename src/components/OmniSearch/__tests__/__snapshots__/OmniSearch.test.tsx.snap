// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OmniSearch renders correctly when dropdown is null 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div />
  .c2 {
  cursor: inherit;
  display: inline-block;
  height: 1.8rem;
  width: 1.8rem;
}

.c0 .rc-dialog-body {
  padding: 7px 0;
}

.c1 {
  padding: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 7px 14px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:focus-within {
  border: 1px solid #5ab2f2;
  box-shadow: 0px 0px 2px #0066ff;
  border-radius: 4px;
}

.c3 {
  width: 100%;
  border: none;
  font-size: 1.6rem;
  margin-left: 1rem;
  height: 2rem;
}

.c3::-webkit-input-placeholder {
  color: #a6a6a6;
}

.c3::-moz-placeholder {
  color: #a6a6a6;
}

.c3:-ms-input-placeholder {
  color: #a6a6a6;
}

.c3::placeholder {
  color: #a6a6a6;
}

<div>
    <div>
      <div
        class="rc-dialog-mask"
      />
      <div
        class="rc-dialog-wrap "
        role="dialog"
        tabindex="-1"
      >
        <div
          class="rc-dialog c0"
          role="document"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden;"
            tabindex="0"
          />
          <div
            class="rc-dialog-content"
          >
            <div
              class="rc-dialog-body"
            >
              <div
                class="c1"
              >
                <span
                  class="c2"
                  fill="#A6A6A6"
                  height="1.8rem"
                  type="oSearch"
                  width="1.8rem"
                >
                  <svg
                    fill="#A6A6A6"
                    height="1.8rem"
                    viewBox="0 0 24 24"
                    width="1.8rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.5 14H14.71L14.43 13.73C15.41 12.59 16 11.11 16 9.5C16 5.91 13.09 3 9.5 3C5.91 3 3 5.91 3 9.5C3 13.09 5.91 16 9.5 16C11.11 16 12.59 15.41 13.73 14.43L14 14.71V15.5L19 20.49L20.49 19L15.5 14ZM9.5 14C7.01 14 5 11.99 5 9.5C5 7.01 7.01 5 9.5 5C11.99 5 14 7.01 14 9.5C14 11.99 11.99 14 9.5 14Z"
                    />
                  </svg>
                </span>
                <input
                  class="c3"
                  placeholder="Search for a report"
                  role="search"
                  type="search"
                  value="demo"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`OmniSearch renders correctly when dropdown is provided 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div />
  .c2 {
  cursor: inherit;
  display: inline-block;
  height: 1.8rem;
  width: 1.8rem;
}

.c0 .rc-dialog-body {
  padding: 7px 0;
}

.c1 {
  padding: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 7px 14px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:focus-within {
  border: 1px solid #5ab2f2;
  box-shadow: 0px 0px 2px #0066ff;
  border-radius: 4px;
}

.c3 {
  width: 100%;
  border: none;
  font-size: 1.6rem;
  margin-left: 1rem;
  height: 2rem;
}

.c3::-webkit-input-placeholder {
  color: #a6a6a6;
}

.c3::-moz-placeholder {
  color: #a6a6a6;
}

.c3:-ms-input-placeholder {
  color: #a6a6a6;
}

.c3::placeholder {
  color: #a6a6a6;
}

<div>
    <div>
      <div
        class="rc-dialog-mask"
      />
      <div
        class="rc-dialog-wrap "
        role="dialog"
        tabindex="-1"
      >
        <div
          class="rc-dialog c0"
          role="document"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden;"
            tabindex="0"
          />
          <div
            class="rc-dialog-content"
          >
            <div
              class="rc-dialog-body"
            >
              <div
                class="c1"
              >
                <span
                  class="c2"
                  fill="#A6A6A6"
                  height="1.8rem"
                  type="oSearch"
                  width="1.8rem"
                >
                  <svg
                    fill="#A6A6A6"
                    height="1.8rem"
                    viewBox="0 0 24 24"
                    width="1.8rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.5 14H14.71L14.43 13.73C15.41 12.59 16 11.11 16 9.5C16 5.91 13.09 3 9.5 3C5.91 3 3 5.91 3 9.5C3 13.09 5.91 16 9.5 16C11.11 16 12.59 15.41 13.73 14.43L14 14.71V15.5L19 20.49L20.49 19L15.5 14ZM9.5 14C7.01 14 5 11.99 5 9.5C5 7.01 7.01 5 9.5 5C11.99 5 14 7.01 14 9.5C14 11.99 11.99 14 9.5 14Z"
                    />
                  </svg>
                </span>
                <input
                  class="c3"
                  placeholder="Search for a report"
                  role="search"
                  type="search"
                  value="demo"
                />
              </div>
              <div
                data-testid="omnisearch-dropdown"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`OmniSearch renders correctly when visible is false 1`] = `
<body>
  <div />
</body>
`;

exports[`OmniSearch renders correctly when visible is true 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div />
  .c2 {
  cursor: inherit;
  display: inline-block;
  height: 1.8rem;
  width: 1.8rem;
}

.c0 .rc-dialog-body {
  padding: 7px 0;
}

.c1 {
  padding: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 7px 14px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:focus-within {
  border: 1px solid #5ab2f2;
  box-shadow: 0px 0px 2px #0066ff;
  border-radius: 4px;
}

.c3 {
  width: 100%;
  border: none;
  font-size: 1.6rem;
  margin-left: 1rem;
  height: 2rem;
}

.c3::-webkit-input-placeholder {
  color: #a6a6a6;
}

.c3::-moz-placeholder {
  color: #a6a6a6;
}

.c3:-ms-input-placeholder {
  color: #a6a6a6;
}

.c3::placeholder {
  color: #a6a6a6;
}

<div>
    <div>
      <div
        class="rc-dialog-mask"
      />
      <div
        class="rc-dialog-wrap "
        role="dialog"
        tabindex="-1"
      >
        <div
          class="rc-dialog c0"
          role="document"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden;"
            tabindex="0"
          />
          <div
            class="rc-dialog-content"
          >
            <div
              class="rc-dialog-body"
            >
              <div
                class="c1"
              >
                <span
                  class="c2"
                  fill="#A6A6A6"
                  height="1.8rem"
                  type="oSearch"
                  width="1.8rem"
                >
                  <svg
                    fill="#A6A6A6"
                    height="1.8rem"
                    viewBox="0 0 24 24"
                    width="1.8rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.5 14H14.71L14.43 13.73C15.41 12.59 16 11.11 16 9.5C16 5.91 13.09 3 9.5 3C5.91 3 3 5.91 3 9.5C3 13.09 5.91 16 9.5 16C11.11 16 12.59 15.41 13.73 14.43L14 14.71V15.5L19 20.49L20.49 19L15.5 14ZM9.5 14C7.01 14 5 11.99 5 9.5C5 7.01 7.01 5 9.5 5C11.99 5 14 7.01 14 9.5C14 11.99 11.99 14 9.5 14Z"
                    />
                  </svg>
                </span>
                <input
                  class="c3"
                  placeholder="Search for a report"
                  role="search"
                  type="search"
                  value="demo"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`OmniSearch uses the value and placeholder props correctly 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div />
  .c2 {
  cursor: inherit;
  display: inline-block;
  height: 1.8rem;
  width: 1.8rem;
}

.c0 .rc-dialog-body {
  padding: 7px 0;
}

.c1 {
  padding: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 7px 14px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:focus-within {
  border: 1px solid #5ab2f2;
  box-shadow: 0px 0px 2px #0066ff;
  border-radius: 4px;
}

.c3 {
  width: 100%;
  border: none;
  font-size: 1.6rem;
  margin-left: 1rem;
  height: 2rem;
}

.c3::-webkit-input-placeholder {
  color: #a6a6a6;
}

.c3::-moz-placeholder {
  color: #a6a6a6;
}

.c3:-ms-input-placeholder {
  color: #a6a6a6;
}

.c3::placeholder {
  color: #a6a6a6;
}

<div>
    <div>
      <div
        class="rc-dialog-mask"
      />
      <div
        class="rc-dialog-wrap "
        role="dialog"
        tabindex="-1"
      >
        <div
          class="rc-dialog c0"
          role="document"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden;"
            tabindex="0"
          />
          <div
            class="rc-dialog-content"
          >
            <div
              class="rc-dialog-body"
            >
              <div
                class="c1"
              >
                <span
                  class="c2"
                  fill="#A6A6A6"
                  height="1.8rem"
                  type="oSearch"
                  width="1.8rem"
                >
                  <svg
                    fill="#A6A6A6"
                    height="1.8rem"
                    viewBox="0 0 24 24"
                    width="1.8rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.5 14H14.71L14.43 13.73C15.41 12.59 16 11.11 16 9.5C16 5.91 13.09 3 9.5 3C5.91 3 3 5.91 3 9.5C3 13.09 5.91 16 9.5 16C11.11 16 12.59 15.41 13.73 14.43L14 14.71V15.5L19 20.49L20.49 19L15.5 14ZM9.5 14C7.01 14 5 11.99 5 9.5C5 7.01 7.01 5 9.5 5C11.99 5 14 7.01 14 9.5C14 11.99 11.99 14 9.5 14Z"
                    />
                  </svg>
                </span>
                <input
                  class="c3"
                  placeholder="placeholder"
                  role="search"
                  type="search"
                  value="hello"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
